react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      letterCounter: 0
+    };
+    this.pwPattern = /\w{8,}/;
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      value: event.target.value,
+      letterCounter: this.state.letterCounter + 1
+    });
```

Comment:
Check it, yo! `letterCounter` is just `event.target.value.length`. So you can omit that state entirely and base your conditional rendering on `this.state.value.length` instead.